src/NSwag.CodeGeneration/Models/OperationModelBase.cs

```diff
@@ -117,7 +117,9 @@ public string UnwrappedResultType
                 }
 
                 var isNullable = response.Value.IsNullable(_settings.CodeGeneratorSettings.SchemaType);
-                return _generator.GetTypeName(response.Value.Schema, isNullable, !response.Value.Schema.HasTypeNameTitle ? "Response" : null);
+                var schemaHasTypeNameTitle = response.Value.Schema?.HasTypeNameTitle;
+                var hint = schemaHasTypeNameTitle.HasValue && !schemaHasTypeNameTitle.Value ? "Response" : null;
```

Comment:
why not `hasTypeNameTitle == false`, using the fact that false != null

Reply:
The exception is in this code segment: response.Value.Schema.HasTypeNameTitle
The problem is that repsonse.Value.Schema is null.
So Schema.HasTypeNameTitle throws a NullReferenceException because the Schema is null.

inside of _generator.GetTypeName its first code segment checks for a null Schema appropriately. Unfortunately to populate the hint parameter, someone used a Schema object to decide the hint without handling the edge case of Schema being null.

Reply:
Yep you can replace `schemaHasTypeNameTitle.HasValue && !schemaHasTypeNameTitle.Value`
with `schemaHasTypeNameTitle == false`

Reply:
i made that change.

Reply:
@RicoSuter resolving comment after that change.